src/modules/System/html_admin/mod_system_settings.html.twig

```diff
@@ -565,9 +570,74 @@ ZW=Zimbabwe
+                    </div>
+                    <div class="card-body">
+                        <div class="alert {% if external_ip %}alert-success{% else %}alert-danger{% endif %}" role="alert" id="external-ip">
+                            {% if external_ip %}
+                                {{ 'With the current settings, your FOSSBilling instance will have an external IP address of: '|trans }} {{ external_ip }}
+                            {% else %}
+                                {{ 'The currently selected network interface does not appear to be able to reach the internet!'|trans }}
+                            {% endif %}
+                        </div>
+
+                        <form class="api-form" method="post" action="{{ 'api/admin/system/set_interface_ip'|link }}" data-api-jsonp="onAfterInterfaceSet">
+                            <input type="hidden" name="CSRFToken" value="{{ CSRFToken }}"/>
+                            <p>{{ 'If your server has multiple network interfaces, you may select the default one for FOSSBilling to use when making requests here.'|trans }}</p>
+                            {% set interface_ips = admin.system_get_interface_ips %}
+                            
+                            <div class="col-12 col-lg-3">
+                                <label for="interface">{{"Select the default network interface:"|trans }}</label>
+                                <select class="form-select" aria-label="Available network interfaces" name="interface" id="interface">
+                                    <option value="0" {% if constant('BIND_TO') == 0 %} selected {%endif%}>{{ 'None (Default PHP Behavior)'|trans }}</option>
+                                    {% for i, interface in interface_ips %}
+                                    <option value="{{ interface }}" {% if constant('BIND_TO') == interface %} selected {% endif %}>{{ interface }}</option>
+                                    {% endfor %}
+                                </select>
+                            </div>
+
+                            <br>
+
+                            <div class="col-12 col-lg-3">
+                                <label for="custom_interface">{{'Enter a custom interface to use:'|trans }}</label>
+                                {% if constant('BIND_TO') not in interface_ips and constant('BIND_TO') != 0%}
+                                    <input class="form-control mt-1" type="text" name="custom_interface" id="custom_interface" value="{{ constant('BIND_TO') }}">
+                                {% else %}
+                                    <input class="form-control mt-1" type="text" name="custom_interface" id="custom_interface">
+                                {% endif %}
+                            </div>
+                            <span class="text-muted">{{ "If the dropdown doesn't contain the correct interface, you can enter the IP address of the correct one above."|trans }}</span>
```

Comment:
I'm sceptical about the functionality to manually set an IP Address. In an edge case, a network IDP which is run by the hoster might catch traffic that goes out over erroneous interface IPs, leading to f.ex. a quarantined vps which results in even more trouble. At the very least, I'd implement something[ like this](https://github.com/FOSSBilling/Proxmox/blob/d61ac9667c9bfdd75770f47cb1d775cefddd6927/src/html_admin/mod_serviceproxmox_ipam.html.twig#L554), to check if the entered IP Address is valid.

Reply:
We don't have a 100% for sure reliable way to fetch all network interfaces, so the ability to manually specify one is somewhat of a necessity.

As for specifically validating if the user did enter an IP address:
- I'm actually relying on the ability to enter something arbitrary here for the live tests as it can then validate that the server is no longer able to make a connection (and therefore indicating that the connection is using the set address). I can think of options to specifically allow this, though.
- Technically, you can also enter the name of a network interface (for example, `eth0`) which will also work correctly. I'll mention this on the page though as I see I haven't.
- It also seems unlikely to me that this could cause issues with network IDP systems. It's trying to bind to a network interface on the server. If it can't bind to it because whatever was specified doesn't exist or is invalid, it'll just produce an error and it'll be entirely unable to make network requests. It's also not like it's going to be trying to scan and find interfaces via brute-force. At most someone would have a typo in the address which results in a few failed connections and then they either fix what they entered or revert it back to the default interface which *should* look just look like typical configuration error since that's what it would be.